### DISCUSSION TITLE
Alias prefix on document type folders

### DISCUSSION THREAD

**Discussion by matthewcare** [umbraco/Umbraco-CMS#12840] [Features and ideas]
Proposal:
When creating folders under "Documnet Types" you have the option of providing an alias prefix.
![image](https://user-images.githubusercontent.com/1469061/184679283-b40aed57-6cc8-49c0-857a-f400d9ace510.png)

For example:
![image](https://user-images.githubusercontent.com/1469061/184679610-8c9cf915-ae79-493e-89fb-12223e31af15.png)
Here "element" is automatically applied to the generated alias.

This prefix will apply to everything created in that folder, sub folders which also supply a value will take presidence.

I personally prefer prefixing the generated classes with thier designation `elementBlaBla`, `compositionBlaBla`, `documentTypeHomePage` to make file structure a little easier to navigate.

I've already experimented with the viability, and have a good proof of concept of it working, but I was wondering if people would actually find the feature useful. If so I'll finish it up, and open a PR.

**Comment by skttl** [comment]
I think it could be useful, but what if you move a type from one folder to another, should it then change the alias to match the new folders prefix?

**Reply by matthewcare** [reply]
Currently once you save a document type the alias no longer auto generates and requires you to manually change it, so I don't think there is any need to handle things like that.
